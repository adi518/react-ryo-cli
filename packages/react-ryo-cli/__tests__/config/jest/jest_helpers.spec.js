const { cloneDeep } = require("lodash");

const { SCRIPTS } = require("../../../lib/constants");
const { overrideJestConfig } = require("../../../config/jest/jest_helpers");

const commonMockArgs = {
  testMatch: [],
  setupFiles: [],
  setupFilesAfterEnv: [],
  moduleNameMapper: {},
  snapshotSerializers: [],
  collectCoverage: null
};

describe("overrideJestConfig", () => {
  test("with build script", () => {
    const mockArgs = [cloneDeep(commonMockArgs), { script: SCRIPTS.BUILD }];
    const result = overrideJestConfig(...mockArgs);

    expect(result).toMatchSnapshot();
  });

  test("with build:development script", () => {
    const mockArgs = [
      cloneDeep(commonMockArgs),
      { script: SCRIPTS.BUILD_DEVELOPMENT }
    ];
    const result = overrideJestConfig(...mockArgs);

    expect(result).toMatchSnapshot();
  });
});
