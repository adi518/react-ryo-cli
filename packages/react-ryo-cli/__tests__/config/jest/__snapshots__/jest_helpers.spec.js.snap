// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overrideJestConfig with build script 1`] = `
Object {
  "collectCoverage": true,
  "moduleNameMapper": Object {
    "\\\\.(css|less|scss|sss|styl)$": "D:\\\\dev\\\\@react-ryo-cli\\\\node_modules\\\\identity-obj-proxy\\\\src\\\\index.js",
  },
  "setupFiles": Array [
    "D:\\\\dev\\\\@react-ryo-cli\\\\packages\\\\react-ryo-cli\\\\config\\\\enzyme\\\\enzyme_setup.js",
  ],
  "setupFilesAfterEnv": Array [
    "D:\\\\dev\\\\@react-ryo-cli\\\\packages\\\\react-ryo-cli\\\\config\\\\jest\\\\jest_setup.js",
  ],
  "snapshotSerializers": Array [
    "enzyme-to-json/serializer",
  ],
  "testMatch": Array [],
}
`;

exports[`overrideJestConfig with build:development script 1`] = `
Object {
  "collectCoverage": false,
  "moduleNameMapper": Object {
    "\\\\.(css|less|scss|sss|styl)$": "D:\\\\dev\\\\@react-ryo-cli\\\\node_modules\\\\identity-obj-proxy\\\\src\\\\index.js",
  },
  "setupFiles": Array [
    "D:\\\\dev\\\\@react-ryo-cli\\\\packages\\\\react-ryo-cli\\\\config\\\\enzyme\\\\enzyme_setup.js",
  ],
  "setupFilesAfterEnv": Array [
    "D:\\\\dev\\\\@react-ryo-cli\\\\packages\\\\react-ryo-cli\\\\config\\\\jest\\\\jest_setup.js",
  ],
  "snapshotSerializers": Array [
    "enzyme-to-json/serializer",
  ],
  "testMatch": Array [],
}
`;
